src/services/refactors/installTypesForPackage.ts

```diff
@@ -46,9 +46,19 @@ namespace ts.refactor.installTypesForPackage {
     function getAction(context: RefactorContext): CodeAction | undefined {
         const { file, startPosition } = context;
         const node = getTokenAtPosition(file, startPosition, /*includeJsDocComment*/ false);
-        if (isStringLiteral(node) && isModuleIdentifier(node) && getResolvedModule(file, node.text) === undefined) {
-            return codefix.tryGetCodeActionForInstallPackageTypes(context.host, file.fileName, node.text);
+        if (!isStringLiteral(node) || !isModuleIdentifier(node)) {
+            return undefined;
         }
+
+        const resolvedTo = getResolvedModule(file, node.text);
+        // Still offer to install types if it resolved to e.g. a ".js" file.
+        // `tryGetCodeActionForInstallPackageTypes` will verify that we're looking for a valid package name,
+        // so the fix won't trigger for imports of ".js" files that couldn't be better replaced by typings.
+        if (resolvedTo && resolvedTo.extension === Extension.Dts) {

```

Comment:
Can it resolve to a ts file and, if so, do we want to suppress the code action in that case?

Reply:
It shouldn't matter because `tryGetCodeActionForInstallPackageTypes` will verify that we're looking for a valid package name, which don't contain `.ts` files when installed, but I've changed this to use `extensionIsTypeScript` anyway since it's just as easy.